### PR TITLE
Add some examples, improve the assembler

### PR DESCRIPTION
**Pull request by jepler**
Assembler improvements:
  * diagnose duplicate labels
  * better diagnostic for jmp to invalid label

Examples:
 * "3 levels of led brightness" from the micropython book
 * "hello" example from pico-examples (turn an LED on and off while needlessly using PIO)

### CODE REVIEW

File: examples/pico-examples/hello.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    out pins, 1
+    jmp loop
+"""
+
+assembled = adafruit_pioasm.assemble(hello)
+
+sm = rp2pio.StateMachine(
+    assembled,
+    frequency=80,
+    init=adafruit_pioasm.assemble("set pindirs 1"),
```
Comment (tannewt) [inline examples/pico-examples/hello.py:25]: You shouldn't need this init or the set pin anymore.

File: examples/pico-examples/hello.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    pull
+    out pins, 1
+    jmp loop
+"""
+
+assembled = adafruit_pioasm.assemble(hello)
+
+sm = rp2pio.StateMachine(
+    assembled,
+    frequency=80,
```
Comment (tannewt) [inline examples/pico-examples/hello.py:24]: The state machine can't go this slow and CP should now raise an error to tell you that.

File: examples/pico-examples/hello.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import board
+import rp2pio
+import adafruit_pioasm
+
+hello = """
+.program hello
+loop:
+    pull
+    out pins, 1
+    jmp loop
```
Comment (tannewt) [inline examples/pico-examples/hello.py:17]: Please add a comment about what this does and note that the jmp loop isn't needed. I'm afraid folks will start with this example and not rely on wrap.

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for these improvements! Just a few comments.

File: adafruit_pioasm.py [hunk 4]
```diff
@@ -85,10 +88,13 @@ def assemble(text_program):
         elif instruction[0] == "jmp":
             #                instr delay cnd addr
             assembled.append(0b000_00000_000_00000)
-            if instruction[-1] in labels:
-                assembled[-1] |= labels[instruction[-1]]
+            target = instruction[-1]
+            if target[:1] in "0123456789":
+                assembled[-1] |= int(target)
```
[the remark this answers is not in the document]
Comment (jepler) [reply adafruit_pioasm.py:93]: Not quite, but maybe the code is too clever as written.  It looks whether the first character (if present) in target is a digit.

**Review by tannewt — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**